internal/provider/provider.go

```diff
@@ -336,64 +354,78 @@ func providerConfigure(p *schema.Provider) schema.ConfigureContextFunc {
-			IDTokenRequestToken: d.Get("oidc_request_token").(string),
-			IDTokenRequestURL:   d.Get("oidc_request_url").(string),
-			IDToken:             d.Get("oidc_token").(string),
-			IDTokenFilePath:     d.Get("oidc_token_file_path").(string),
-
-			// Feature Toggles
-			SupportsClientCertAuth:         true,
-			SupportsClientSecretAuth:       true,
-			SupportsOIDCAuth:               d.Get("use_oidc").(bool),
-			SupportsManagedServiceIdentity: d.Get("use_msi").(bool),
-			SupportsAzureCliToken:          true,
-			SupportsAuxiliaryTenants:       len(auxTenants) > 0,
-
-			// Doc Links
-			ClientSecretDocsLink: "https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_client_secret",
-
-			// Use MSAL
-			UseMicrosoftGraph: true,
+		var clientCertificateData []byte
+		if encodedCert := d.Get("client_certificate").(string); encodedCert != "" {
+			var err error
+			clientCertificateData, err = decodeCertificate(encodedCert)
+			if err != nil {
+				return nil, diag.FromErr(err)
+			}
 		}
 
-		config, err := builder.Build()
-		if err != nil {
-			return nil, diag.Errorf("building AzureRM Client: %s", err)
+		var (
+			env *environments.Environment
+			err error
+
+			envName      = d.Get("environment").(string)
+			metadataHost = d.Get("metadata_host").(string)
+		)
+
+		if metadataHost != "" {
+			if env, err = environments.FromEndpoint(ctx, fmt.Sprintf("https://%s", metadataHost), envName); err != nil {
+				return nil, diag.FromErr(err)
+			}
+		} else if env, err = environments.FromName(envName); err != nil {
+			return nil, diag.FromErr(err)
 		}
 
-		terraformVersion := p.TerraformVersion
-		if terraformVersion == "" {
-			// Terraform 0.12 introduced this field to the protocol
-			// We can therefore assume that if it's missing it's 0.10 or 0.11
-			terraformVersion = "0.11+compatible"
+		var (
+			enableAzureCli        = d.Get("use_cli").(bool)
+			enableManagedIdentity = d.Get("use_msi").(bool)
+			enableOidc            = d.Get("use_oidc").(bool)
+		)
+
+		authConfig := auth.Credentials{
+			Environment:        *env,
+			ClientID:           d.Get("client_id").(string),
+			TenantID:           d.Get("tenant_id").(string),
+			AuxiliaryTenantIDs: auxTenants,
+
+			ClientCertificateData:     clientCertificateData,
+			ClientCertificatePath:     d.Get("client_certificate_path").(string),
+			ClientCertificatePassword: d.Get("client_certificate_password").(string),
+			ClientSecret:              d.Get("client_secret").(string),
+
+			OIDCAssertionToken:          d.Get("oidc_token").(string),
+			GitHubOIDCTokenRequestURL:   d.Get("oidc_request_url").(string),
+			GitHubOIDCTokenRequestToken: d.Get("oidc_request_token").(string),
+
+			CustomManagedIdentityEndpoint: d.Get("msi_endpoint").(string),
+
+			EnableAuthenticatingUsingClientCertificate: true,
+			EnableAuthenticatingUsingClientSecret:      true,
+			EnableAuthenticatingUsingAzureCLI:          enableAzureCli,
```

Comment:
AzureAD has had this property for awhile, the intention is to provide a means of disabling CLI auth in noninteractive environments, which helps with error messages since instead of getting a failure mentioning Azure CLI, you'd get an error stating that no authentication methods could be configured.